docs/building_pipelines.rst

```diff
@@ -293,3 +294,423 @@ Warnings raised will also be logged to the error log but will not cause data to
+
+The data returned from the *attach* command is the status of running that command encoded within a JSON document.
+
+At this point the debugger is attached and collecting data at each node within the pipeline. By default only one reading is retained at each point in the pipeline; this is the last reading seen at that point in the pipeline.
+
+Buffer
+######
+
+The buffer command will display the data at each node in the pipeline. In the example pipeline used here there are several nodes, including a branch in the pipeline.
+
++-----------------+
+| |LathePipeline| |
++-----------------+
+
+The output of the Lathe simulator south plugin is split into two branches of the pipeline. An asset filter is used at the start of each branch to control which assets are sent down that branch of the pipeline. In this case the *latheCurrent* readings are sent down one branch and all other readings are sent down the other branch.
+
+Running the *buffer* command will display the data recorded on the input of each node in the pipeline.
+
+.. code-block:: bash
+
+   Debug: Lathe$ buffer
+   {
+     "data": [
+       {
+         "name": "Branch",
+         "readings": [
+           {
+             "asset_code": "latheIR",
+             "user_ts": "2025-03-31 16:23:10.008024+00:00",
+             "ts": "2025-03-31 16:23:10.008024+00:00",
+             "reading": {
+               "gearbox": 27.93525,
+               "motor": 29.6282,
+               "headstock": 25.51,
+               "tailstock": 20.52,
+               "tool": 24.0905
+             }
+           }
+         ]
+       },
+       [
+         {
+           "name": "CurrentOnly",
+           "readings": [
+             {
+               "asset_code": "latheIR",
+               "user_ts": "2025-03-31 16:23:10.008024+00:00",
+               "ts": "2025-03-31 16:23:10.008024+00:00",
+               "reading": {
+                 "gearbox": 27.93525,
+                 "motor": 29.6282,
+                 "headstock": 25.51,
+                 "tailstock": 20.52,
+                 "tool": 24.0905
+               }
+             }
+           ]
+         },
+         {
+           "name": "RMS",
+           "readings": [
+             {
+               "asset_code": "latheCurrent",
+               "user_ts": "2025-03-31 16:23:10.008018+00:00",
+               "ts": "2025-03-31 16:23:10.008018+00:00",
+               "reading": {
+                 "current": 767
+               }
+             }
+           ]
+         },
+         {
+           "name": "Writer",
+           "readings": [
+             {
+               "asset_code": "latheCurrent RMS",
+               "user_ts": "2025-03-31 16:23:02.803881+00:00",
+               "ts": "2025-03-31 16:23:02.803881+00:00",
+               "reading": {
+                 "current": 773.9413414465,
+                 "currentpeak": 47,
+                 "currentcrest": 0.0607281166
+               }
+             }
+           ]
+         }
+       ],
+       {
```

Comment:
We should have it an array, even if a single object in that. This is making JSON hard to parse and looks confusing.